.github/scripts/fetch_compare_version.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2019 Intel Corporation.  All rights reserved.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+
+import re
+import shlex
+import subprocess
+import sys
+
+
+def fetch_version_from_code():
+    command = 'egrep "set *\(WAMR_VERSION" build-scripts/config_common.cmake'
+    cp = subprocess.run(shlex.split(command), capture_output=True, check=True)
+
+    output = cp.stdout.decode()
+    if len(output) == 0:
+        raise Exception("Unexpected output")
+
+    major_match = re.search("WAMR_VERSION_MAJOR (\d+)", output)
+    if major_match is None:
+        raise Exception("Unexpected patter of VERSION_MAJOR")
+    major = major_match.groups()[0]
+
+    minor_match = re.search("WAMR_VERSION_MINOR (\d+)", output)
+    if minor_match is None:
+        raise Exception("Unexpected patter of VERSION_MINOR")
+    minor = minor_match.groups()[0]
+
+    patch_match = re.search("WAMR_VERSION_PATCH (\d+)", output)
+    if patch_match is None:
+        raise Exception("Unexpected patter of VERSION_PATCH")
+    patch = patch_match.groups()[0]
+
+    return f"WAMR-{major}.{minor}.{patch}"
+
+
+def fetch_latest_git_tag():
+    list_tag_cmd = (
+        'git tag --list WAMR-*.*.* --sort=committerdate --format="%(refname:short)"'
+    )
+    cp = subprocess.run(shlex.split(list_tag_cmd), capture_output=True, check=True)
+
+    all_tags = cp.stdout.decode().strip()
+    latest_tag = all_tags.split("\n")[-1]
+    return latest_tag
+
+
+def match_version_pattern(v):
+    pattern = "WAMR-\d+\.\d+\.\d+"
+    m = re.match(pattern, v)
+    return m is not None
+
+
+def split_version_string(v):
+    """
+    return the semantic version as an integer tuple
+    """
+    pattern = "WAMR-(\d+)\.(\d+)\.(\d+)"
+    m = re.match(pattern, v)
+    return (int(x) for x in m.groups())
+
+
+def compare_version_string(v1, v2):
+    """
+    return value:
+      - 1. if v1 > v2
+      - -1. if v1 < v2
+      - 0. if v1 == v2
+    """
+    if not match_version_pattern(v1):
+        raise Exception(f"{v1} doesn't match the version pattern")
+
+    if not match_version_pattern(v2):
+        raise Exception(f"{v2} doesn't match the version pattern")
+
+    v1_major, v1_minor, v1_patch = split_version_string(v1)
+    v2_major, v2_minor, v2_patch = split_version_string(v2)
+
+    if v1_major > v2_major:
+        return 1
+    elif v1_major < v2_major:
+        return -1
+    else:
+        if v1_minor > v2_minor:
+            return 1
+        elif v1_minor < v2_minor:
+            return -1
+        else:
+            if v1_patch > v2_patch:
+                return 1
+            elif v1_patch < v2_patch:
+                return -1
+            else:
+                return 0
```

Comment:
i guess it's simpler to compare tuples.
```
>>> (1,2,1) > (1,1,2)
True
>>> 
```
https://docs.python.org/3/reference/expressions.html#value-comparisons

Reply:
It is a really good one